### PR TITLE
add toggle buttons to pub field form elements

### PR DESCRIPTION
**Pull request by 3mcd**
## Issue(s) Resolved

Closes https://github.com/pubpub/pubpub/issues/3179

(Looks like this issue was accidentally created in the pubpub/pubpub repo)

## High-level Explanation of PR

This PR adds a button to the left side of form elements within the pub create form. Clicking the button toggles the disabled state for that field. By default all fields are enabled.

Disabled fields are omitted from the form when submitted. This means that submitting a form with all fields disabled will create a pub with no pub values.

## Test Plan

1. Use the create pub button to create a pub.
2. Create pubs with various configurations of disabled and enabled fields. Values of disabled fields should not be present on newly created pubs.
3. The form's submit button should only be disabled when _enabled_ buttons fail to validate. Invalid _disabled_ fields should not be considered.
4. Update a pub. The buttons should not be visible on the update form.

## Screenshots (if applicable)

<figure>
  <figcaption>The pub creation form with a disabled title field and invalid email field. The "Create Pub" button is disabled because of the enabled, invalid field.</figcaption>
  <p align="center">
    Screenshot 2024-09-26 at 1 40 43 PM
  </p>
</figure>

<figure>
  <figcaption>The same pub creation form after disabling the invalid email field. The "Create Pub" button is now enabled because the invalid field was disabled.</figcaption>
  <p align="center">
    Screenshot 2024-09-26 at 1 40 49 PM
  </p>
</figure>

<figure>
  <figcaption>The pub page of the newly created pub. Notice that the only field associated with the pub is "Member ID": the only enabled field on the form in the previous screenshot.
  <p align="center">
    Screenshot 2024-09-26 at 1 41 12 PM
  </p>
</figure>

### CODE REVIEW

File: _tmp_68517_284c3f327395000692a48c4688f69f96
Comment (allisonking) [inline _tmp_68517_284c3f327395000692a48c4688f69f96:1]: not needed?

**Review by allisonking — APPROVED** [review]
this looks good for the most part! just two things I found while playing around that may not be need to be addressed in this PR:

1. I can't create a pub with all disabled fields (I get a db error)

```
[16:15:08.231] ERROR: syntax error at or near ")"
    package: "core"
    err: {
      "type": "DatabaseError",
      "message": "syntax error at or near \")\"",
      "stack":
          error: syntax error at or near ")"
              at /Users/allisonking/workspace/kf/v7/node_modules/.pnpm/pg@8.12.0/node_modules/pg/lib/client.js:526:17
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at async PostgresConnection.executeQuery (/Users/allisonking/workspace/kf/v7/core/.next/server/chunks/ssr/33c0b_kysely_dist_esm_85a464._.js:3973:28)
```

2. The create form doesn't seem to close after submitting (not sure if that was introduced in this PR)

File: _tmp_68517_284c3f327395000692a48c4688f69f96
Comment (3mcd) [reply _tmp_68517_284c3f327395000692a48c4688f69f96:1]: uhh yeah not sure where this came from, ty!

File: core/lib/server/pub.ts [hunk 1]
```diff
@@ -86,9 +86,9 @@ const pubValues = (
 	const alias = "latest_values";
 	// Although kysely has selectNoFrom, this kind of query can't be generated without using raw sql
 	const jsonObjAgg = (subquery: AliasedSelectQueryBuilder<any, any>) =>
-		sql<PubValues>`(select json_object_agg(${sql.ref(alias)}.slug, ${sql.ref(
+		sql<PubValues>`(select coalesce(json_object_agg(${sql.ref(alias)}.slug, ${sql.ref(
 			alias
-		)}.value) from ${subquery})`;
+		)}.value), '{}') from ${subquery})`;
```
Comment (3mcd) [inline core/lib/server/pub.ts:91]: This just coalesces the value to `{}` in the case the pub has no pub values.

**Pull request merged**